### PR TITLE
Azure cloud settings from Grafana Azure SDK

### PR DESCRIPTION
**Pull request by kostrse**
This PR demonstrates use of Azure cloud settings from the Grafana Azure SDK.

See https://github.com/grafana/grafana-azure-sdk-go/pull/78.

### CODE REVIEW

**Review by aangelisc — CHANGES_REQUESTED** [review]
I think we only need the backend changes here for now. I'd much prefer to define the appropriate settings in the frontend SDK also to avoid adding them to the Grafana config.

File: pkg/api/dtos/frontend_settings.go [hunk 1]
```diff
@@ -56,11 +56,17 @@ type FrontendSettingsLicenseInfoDTO struct {
...
+	DisplayName string `json:"displayName"`
+}
+
 type FrontendSettingsAzureDTO struct {
-	Cloud                   string `json:"cloud"`
-	ManagedIdentityEnabled  bool   `json:"managedIdentityEnabled"`
-	WorkloadIdentityEnabled bool   `json:"workloadIdentityEnabled"`
-	UserIdentityEnabled     bool   `json:"userIdentityEnabled"`
+	Cloud                   string                          `json:"cloud"`
+	Clouds                  []FrontendSettingsAzureCloudDTO `json:"clouds"`
```
[the remark this answers is not in the document]
Comment (kostrse) [reply pkg/api/dtos/frontend_settings.go:66]: This is not the full cloud configuration, just list of IDs and display names ([here](https://github.com/grafana/grafana/blob/fc0742a0bc48c7abae759eb9a144cfb8988a97ed/pkg/api/dtos/frontend_settings.go#L59-L62)).
And it's being loaded once upon initialization.

Also the same structures will be removed from datasources.

**Comment by kostrse** [comment]
> I think we only need the backend changes here for now. I'd much prefer to define the appropriate settings in the frontend SDK also to avoid adding them to the Grafana config.

Could you please propose a solution, how frontend would be able to know list of clouds?

**Comment by aangelisc** [comment]
> Could you please propose a solution, how frontend SDK would be able to know list of clouds?

@kostrse they should be hardcoded. Whilst this will lead to additional maintenance as we will have the list of clouds in two places, it removes an unnecessary piece of data from the Grafana settings. As these constants are specific to Azure data sources the rest of Grafana does not need to be aware of them.

The Azure cloud configuration doesn't change frequently enough where this would be a substantial maintenance burden.

**Comment by kostrse** [comment]
This PR was pas a prototype, it has been replaced by:
* #83460
* #85415

**Pull request closed without merging**